packages/data-stores/src/i18n/index.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { controls } from '@wordpress/data-controls';
+import { registerStore } from '@wordpress/data';
+import type { SelectFromMap, DispatchFromMap } from '../mapped-types';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from './constants';
+import reducer, { State } from './reducer';
+import * as actions from './actions';
+import * as selectors from './selectors';
+import * as resolvers from './resolvers';
+
+export type { State };
+export type { LocalizedLanguageNames } from './types';
+
+let isRegistered = false;
+
+export function register(): typeof STORE_KEY {
+	if ( ! isRegistered ) {
+		isRegistered = true;
+		registerStore< State >( STORE_KEY, {
+			resolvers,
+			actions,
+			controls,
+			reducer: reducer as any,
+			selectors,
+		} );
+	}
+	return STORE_KEY;
+}
+
+declare module '@wordpress/data' {
```

Comment:
This seems to be a recurring pattern; I wonder if we could share this ambient module with the rest of the data stores in this package.

Reply:
@sirreal knows more about this than I do, but from what I understand this is going to be required to be able to continue overloading the `select` and `dispatch` functions for each store.

Reply:
Makes sense. I was just wondering if there's a way to share that declaration vs doing it in every store again and again.

Reply:
As far as I know, there's no way to improve the declaration; it has to be top level. The insides are as good as we could get them but most things have room for improvement 🙂 

When the data package is typed we may be able to improve this.

There's also a recent change (https://github.com/WordPress/gutenberg/pull/26655) where the the store instances are passed in instead of string keys. This has very interesting implications that could make these declarations totally redundant!